app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionaryAddWordFragment.java

```diff
@@ -0,0 +1,253 @@
+
+// Caveat: This class is basically taken from
+// packages/apps/Settings/src/com/android/settings/inputmethod/UserDictionaryAddWordFragment.java
+// in order to deal with some devices that have issues with the user dictionary handling
+
+/**
+ * Fragment to add a word/shortcut to the user dictionary.
+ * As opposed to the UserDictionaryActivity, this is only invoked within Settings
+ * from the UserDictionarySettings.
+ */
+public class UserDictionaryAddWordFragment extends SubScreenFragment {
+    // TODO: To be reactivated when UserDictionaryLocalePicker.UserDictionaryLocalePicker() is implemented
+    //implements LocationChangedListener
+
+    private UserDictionaryAddWordContents mContents;
+    private View mRootView;
+    private EditText mWordEditText;
+    private EditText mWeightEditText;
+    private InputMethodManager mInput;
+    private ActionBar mActionBar;
+
+    @NonNull
+    @Override
+    public View onCreateView(final LayoutInflater inflater, final ViewGroup container,
+            final Bundle savedState) {
+        mRootView = inflater.inflate(R.layout.user_dictionary_add_word_fullscreen, null);
+        // If we have a non-null mContents object, it's the old value before a configuration
+        // change (eg rotation) so we need to use its values. Otherwise, read from the arguments.
+        if (null == mContents) {
+            mContents = new UserDictionaryAddWordContents(mRootView, getArguments());
+        } else {
+            // We create a new mContents object to account for the new situation : a word has
+            // been added to the user dictionary when we started rotating, and we are now editing
+            // it. That means in particular if the word undergoes any change, the old version should
+            // be updated, so the mContents object needs to switch to EDIT mode if it was in
+            // INSERT mode.
+            mContents = new UserDictionaryAddWordContents(mRootView, mContents);
+        }
+
+        mWordEditText = mRootView.findViewById(R.id.user_dictionary_add_word_text);
+        mWeightEditText = mRootView.findViewById(R.id.user_dictionary_add_weight);
+
+        final Bundle args = getArguments();
+        mActionBar = ((AppCompatActivity) requireActivity()).getSupportActionBar();
+        if (args != null) {
+            if (args.getInt(UserDictionaryAddWordContents.EXTRA_MODE) == UserDictionaryAddWordContents.MODE_EDIT) {
+                mActionBar.setTitle(R.string.user_dict_settings_edit_dialog_title);
+            } else {
+                mActionBar.setTitle(R.string.user_dict_settings_add_dialog_title);
+            }
+            mActionBar.setSubtitle(UserDictionarySettings.getLocaleDisplayName(getActivity(), mContents.getDropDownMenuLanguage()));
+        }
+
+        final Button saveWordButton = mRootView.findViewById(R.id.user_dictionary_save_button);
+        final Drawable saveWordIcon = getBitmapFromVectorDrawable(R.drawable.ic_save, 1f);
```

Comment:
Is the conversion to a bitmap drawable necessary even if scale is 1?

Reply:
Btw you use a button and set a drawable in code. Is there any reason why this is preferable over an `ImageButton` and setting the drawable in the xml file?

Reply:
Oops! 😬 Bad copy/paste without thinking. Fixed with commit 9851488